setup.py

```diff
@@ -28,20 +47,64 @@ def proto_compile(output_path):
             file.truncate()
 
 
-class ProtoCompileInstall(install):
+def libp2p_build_install():
+    try:
+        proc = subprocess.Popen(['go', 'version'], stdout=subprocess.PIPE)
+        result, _ = proc.communicate()
+        result = result.decode('ascii', 'replace')
+        m = re.search(r'^go version go([\d.]+)', result)
+        v = m.group(1)
+
+        if version.parse(v) < version.parse("1.13"):
+            raise EnvironmentError(f'newer version of go required: must be >= 1.13, found {version}')
+
+    except FileNotFoundError:
+        raise FileNotFoundError('could not find golang installation')
+
+    with tempfile.TemporaryDirectory() as tempdir:
+        url = f'https://github.com/learning-at-home/go-libp2p-daemon/archive/refs/tags/{P2PD_VERSION}.tar.gz'
+        dest = os.path.join(tempdir, 'libp2p-daemon.tar.gz')
+        urllib.request.urlretrieve(url, dest)
+
+        tar = tarfile.open(dest, 'r:gz')
+        tar.extractall(tempdir)
+        tar.close()
+
+        result = subprocess.run(['go', 'build', '-o', os.path.join(here, "hivemind", "hivemind_cli", "p2pd")],
+                                cwd=os.path.join(tempdir, f'go-libp2p-daemon-{P2PD_VERSION[1:]}', 'p2pd'))
```

Comment:
Also, it's more readable to use shell=True and one statement for the command instead of multiple arguments in a list

Reply:
Please use `f'go build -o {shlex.quote(os.path.join(here, "hivemind", "hivemind_cli", "p2pd"))}'`.

Without the `shlex.quote`, running commands with `shell=True` may lead to a security [vulnerability](https://docs.python.org/3/library/shlex.html#shlex.quote) (maybe not in this case, but it's a good practice to write secure code everywhere since the code from one place may be unexpectedly reused in others).